src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -102,7 +107,10 @@ public IEnumerable<string> GetManifestDirectories()
 
                 foreach (var workloadManifestDirectory in directoriesWithManifests.Values)
                 {
-                    yield return workloadManifestDirectory;
+                    if (!_preview4ManifestIds.Any(prev4Manifest => workloadManifestDirectory.ToLowerInvariant().Contains(prev4Manifest.ToLowerInvariant())))
```

Comment:
Logic should be changed here too.  Perhaps it would be a good idea to factor the logic out into a helper method so it's not repeated.